motion_tracker/utils/image_generator.py

```diff
@@ -0,0 +1,151 @@
+"""A script for building image generators for net modeling."""
+
+import pandas as pd
+import numpy as np
+import cv2
+from image_cropping import Coords, crop_and_resize
+from crop_vis import show_single_stage 
+
+def master_generator(num_video_crops=10, num_image_crops=10, batch_size=50): 
+    """Generator yielding a batch of images. 
+
+    Args: 
+    ----
+        num_video_crops: int
+            Number of random crops of the current frame to take. 
+            Denoted as k3 in the paper, with a value of 10. 
+        num_image_crops: int 
+            Number of random crops of the randomly sampled image 
+            to take. Dentoed as k4 in the paper, with a value of 10. 
+        batch_size: int
+            Number of items in list to be returned. 
+    """
+    # To switch witch training images we draw from first in 
+    # a batch. 
+    imagenet = True
+    imagenet_gen = imagenet_generator(num_image_crops)
+    alov_gen = alov_generator(num_video_crops)
+
+    while True: 
+        
+        out_imgs = []
+        out_boxes = []
+        video_pairs = 0
+        images = 0
+        num_obs = 0
+        # Might need to add in some check or something in the 
```

Comment:
While it's possible one of the groups of crops will get cut off, that doesn't seem like a big deal, and I think the current code implements this well enough.